examples/src/main/java/com/example/JavaFuzzTest.java

```diff
@@ -0,0 +1,24 @@
+package com.example;
```

Comment:
What is the role of the `src/main/java/com/example` directory structure? Is it mandated by Bazel? (I'm not quite familiar with Java rules.)

If this is mandatory, I was going to suggest to place the Java code under an `examples/java` directory. I.e., this would become `examples/java/src/...`. This way, it will be clear why some code requires this nesting structure.

It might also make sense to place the Java fuzz targets under `examples/java`, but I'm fine either way.

Reply:
While not required, it is certainly recommended: https://docs.bazel.build/versions/master/tutorial/java.html

I like the idea of moving the tests to a `java` subpackage. 

Should I leave the C targets as is or move them into a `cc` package?

Reply:
> While not required, it is certainly recommended: https://docs.bazel.build/versions/master/tutorial/java.html
> 
> I like the idea of moving the tests to a `java` subpackage.

Since we also need a different BUILD target naming convention, I'm becoming more in favor of this alternative too.

> 
> Should I leave the C targets as is or move them into a `cc` package?

If this was done from the beginning, that would have made sense, but now I feel it would be disruptive for anyone already using the library and expecting to find these examples here (e.g., in scripts). So we can leave them here for now and reconsider later if needed.

Reply:
On second thought, moving the `java_fuzz_test`s into `java` directory messes up the `target_class` detection. Can I move them to `examples_java` instead, leaving the `cc_fuzz_test`s in `example`?

Reply:
> On second thought, moving the `java_fuzz_test`s into `java` directory messes up the `target_class` detection. 

I'm not sure I follow this part. How does the base path of the BUILD file make a difference? I.e., does it matter the Java tree is rooted at `examples/{BUILD,src/main/...}` vs. `examples/java/{BUILD,src/main/...}`?

Reply:
It does. The heuristic for determining the main class does scan the full path for the first occurence of `java` or `src` (see https://github.com/bazelbuild/bazel/blob/578a72c8448c21ecdb6ea7585b9da348fa1d7fd5/src/main/java/com/google/devtools/build/lib/rules/java/JavaCommon.java#L504).

We could of course just specify the `target_class` attribute in the examples, but that wouldn't tell users that they can expect the same semantics as from `java_binary`.

Reply:
I found a way to please the heuristic and move everything into `examples/java`.